deepchecks/utils/abstracts/confusion_matrix_abstract.py

```diff
@@ -51,38 +57,69 @@ def create_confusion_matrix_figure(confusion_matrix_data: np.ndarray, classes_na
-        text=confusion_matrix_data, texttemplate=text_template))
-    fig.data[0].colorbar.title = color_bar_title
-    fig.update_layout(title=plot_title)
-    fig.update_layout(height=600)
-    fig.update_xaxes(title='Predicted Value', type='category', scaleanchor='y', constrain='domain')
-    fig.update_yaxes(title='True Value', type='category', constrain='domain', autorange='reversed')
-
-    return fig
+    display = []
+    confusion_matrix_norm = confusion_matrix_data.astype('float') / \
+        (confusion_matrix_data.sum(axis=1)[:, np.newaxis] + np.finfo(float).eps) * 100
+
+    accuracy_array = np.diag(confusion_matrix_norm).round(decimals=2)
+    sorted_map = dict(sorted(dict(enumerate(accuracy_array)).items(), key=lambda item: item[1]))
+
+    worst_class_indices = [index for index, acc in sorted_map.items() if acc < MIN_ACCURACY_FOR_GOOD_CLASSES]
+
+    if len(accuracy_array) > MAX_WORST_CLASSES_TO_DISPLAY:
+        worst_class_indices = worst_class_indices[:MAX_WORST_CLASSES_TO_DISPLAY]
+
+    display_msg = textwrap.dedent(
+        f'The overall accuracy of your model is: {round(np.sum(accuracy_array)/len(accuracy_array), 2)}%.<br>'
+        f'Best accuracy achieved on samples with <b>{classes_names[np.argmax(accuracy_array)]}</b> '
+        f'label ({np.max(accuracy_array)}%).<br>Worst accuracy achieved on samples with '
+        f'<b>{classes_names[np.argmin(accuracy_array)]}</b> label ({np.min(accuracy_array)}%).<br>'
+        'Below are pie charts showing the prediction distribution for samples grouped based on their label.'
+    )
+    display.append(display_msg)
+    curr_col = 1
+    if len(worst_class_indices) > 0:
+        fig = make_subplots(rows=1, cols=min(MAX_WORST_CLASSES_TO_DISPLAY, len(worst_class_indices)),
+                            specs=[[{'type': 'pie'}] * min(MAX_WORST_CLASSES_TO_DISPLAY, len(worst_class_indices))])
+        total_samples = np.sum(confusion_matrix_data, axis=None)
+        for idx in worst_class_indices:
+            values = np.delete(confusion_matrix_data[idx], idx)
+            n_samples_with_label = np.sum(confusion_matrix_data[idx])
+            label_percentage = format_percent(n_samples_with_label/total_samples)
+            labels = np.delete(classes_names, idx)
+            if len(values) > max_num_labels_to_show:
+                sorted_indices = np.argsort(values)[::-1]
+                values = values[sorted_indices]
+                labels = labels[sorted_indices]
+                other_labels_sum = np.sum(values[max_num_labels_to_show:])
+                other_label_names = ', '.join(labels[max_num_labels_to_show:])
+                labels = labels[:max_num_labels_to_show]
+                values = values[:max_num_labels_to_show]
+                values = np.append(values, other_labels_sum)
+                labels = np.append(labels, 'Others')
+
+            fig.add_trace(go.Pie(values=values, title=f'Label: <b>{classes_names[idx]}</b><br>'
```

Comment:
Would make the header for each plot:
** Anger ** (can use html commends like header / sub-header)
'# Samples': X, 'Accurate predictions':  Y (Z%) (try same line, liked the bold you did on the values)